libs/generic-view/ui/src/lib/predefined/data-migration/data-migration.tsx

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+/* stylelint-disable no-duplicate-selectors */
+
+import React, { FunctionComponent, useEffect } from "react"
+import { APIFC } from "generic-view/utils"
+import { McDataMigrationConfig } from "generic-view/models"
+import { useDispatch, useSelector } from "react-redux"
+import {
+  selectDataMigrationSourceDevice,
+  selectDataMigrationSourceDevices,
+  selectDataMigrationTargetDevices,
+  setSourceDevice,
+} from "generic-view/store"
+import { Instruction, InstructionWrapper } from "./instruction"
+import styled from "styled-components"
+import { H3 } from "../../texts/headers"
+import { Divider } from "../../helpers/divider"
+import { TargetSelector, TargetSelectorWrapper } from "./target-selector"
+import Form from "../../interactive/form/form"
+import { TransferSetup } from "./transfer-setup"
+import { GenericThemeProvider } from "generic-view/theme"
+import { Device } from "./components/device-card"
+import { getActiveDevice } from "Core/device-manager/selectors/get-active-device.selector"
+import { defineMessages } from "react-intl"
+import { intl } from "Core/__deprecated__/renderer/utils/intl"
+
+const messages = defineMessages({
+  header: {
+    id: "module.genericViews.dataMigration.header",
+  },
+})
+
+const DataMigrationUI: FunctionComponent<McDataMigrationConfig> = ({
+  dataTypes,
+}) => {
+  const dispatch = useDispatch()
+  const activeDevice = useSelector(getActiveDevice)
+  const sourceDevices = useSelector(
+    selectDataMigrationSourceDevices
+  ) as Device[]
+  const targetDevices = useSelector(
+    selectDataMigrationTargetDevices
+  ) as Device[]
+  const sourceDevice = useSelector(selectDataMigrationSourceDevice)
+
+  const singleDeviceConnected =
```

Comment:
Two things. 
Please ensure that the condition is correct. Looks a little bit suspicious. Consider a case when there are two source devices and zero target devices, and when there are two target devices and zero source devices. 
Second thing. The only usage of singleDeviceConnected is located 2 lines below, and it is a renaming to displayInstruction. Maybe instead of that use displayInstructionForSingleDevice as a name in the first place.

Reply:
Fixed in [3475e46](https://github.com/mudita/mudita-center/pull/1885/commits/3475e4620108ac3f16963af7255fb1b7e72b8f78)